i3lock.c

```diff
@@ -1526,6 +1725,17 @@ bool load_slideshow_images(const char *path) {
     return true;
 }
 
+void gif_anim_loop(struct ev_loop *loop, struct ev_timer *timer, int delay) {
+    static int img_count = 0;
+
+    if (++img_count >= gif_img_count) img_count = 0;
+    img = gif_img[img_count].img;
+    redraw_screen();
+    ev_timer_stop(loop, timer);
```

Comment:
If `redraw_screen` takes non-negligible time, should this be run before or after it?

Reply:
In that particular case, as I understand, it doesn't matter. It is because we need to restart the timer explicitly each time we enter into the callback. But, probably, it would be better to stop the timer before redraw as it confuse less when reading the code.